scripts/zkai_native_attention_mlp_rmsnorm_adjacent_layout_gate.py

```
@@ -0,0 +1,581 @@
+    return results
+
+
+def validate_mutation_results(results: Any) -> None:
+    if not isinstance(results, list):
+        raise AdjacentLayoutGateError("mutation results must be a list")
+    names = [entry.get("name") for entry in results if isinstance(entry, dict)]
+    if names != [name for name, _ in MUTATIONS]:
+        raise AdjacentLayoutGateError("mutation inventory drift")
+    for entry in results:
+        if set(entry) != {"name", "rejected", "reason"}:
+            raise AdjacentLayoutGateError("mutation entry key drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)